batch/batch/driver/instance_collection_manager.py

```diff
@@ -16,10 +17,11 @@ def __init__(self, app, machine_name_prefix):
         self.db: Database = app['db']
         self.zone_monitor: ZoneMonitor = app['zone_monitor']
         self.machine_name_prefix = machine_name_prefix
-        self.inst_coll_regex = re.compile(f'{self.machine_name_prefix}(?P<inst_coll>[^-]+)-.*')
+        self.inst_coll_regex = re.compile(f'{self.machine_name_prefix}(?P<inst_coll>.*)-.*')
 
         self.name_inst_coll = {}
         self.name_pool = {}
+        self.name_job_private = {}
 
     async def async_init(self):
         inst_coll_records = self.db.execute_and_fetchall('''
```

Comment:
I feel like we should just get all the necessary information in one DB select? We can `LEFT JOIN` onto the pools table to get the necessary information (or NULL in the JPIM case) and pass it all into the Pool.